### PR TITLE
Generically update expired apt keys

### PR DESCRIPTION
**Pull request by joepvd**
And limit the mongo apt repos to arch=amd64

https://admin-staging.travis-ci.org/build/696936

### CODE REVIEW

File: lib/travis/build/appliances/update_apt_keys.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'travis/build/appliances/base'
+
+module Travis
+  module Build
+    module Appliances
+      class UpdateAptKeys < Base
+        def apply
+          command = <<-EOF
+          if command -v apt-get &>/dev/null && [[ -d /var/lib/apt/lists ]]; then
+            LANG=C apt-key list | awk -F'[ /]+' '/expired:/{printf "apt-key adv --recv-keys --keyserver keys.gnupg.net %s\\n", $3}' | sudo sh &>/dev/null
```
Comment (BanzaiMan) [inline lib/travis/build/appliances/update_apt_keys.rb]: Is it OK to assume that keys.gnupg.net has the new key?

File: lib/travis/build/appliances/update_mongo_arch.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+require 'travis/build/appliances/base'
+
+module Travis
+  module Build
+    module Appliances
+      class UpdateMongoArch < Base
+        def apply
+          command = <<-EOF
+            if command -v lsb_release &>/dev/null && [[ $(lsb_release -sc) != precise ]]; then
```
Comment (BanzaiMan) [inline lib/travis/build/appliances/update_mongo_arch.rb]: Why is Precise excluded from this list? (I'm not saying that we should keep supporting Precise, but that this extra condition seems out of place.)

File: lib/travis/build/appliances/update_apt_keys.rb (hunk 1 above)
Comment (joepvd) [reply lib/travis/build/appliances/update_apt_keys.rb]: I believe it is a reasonable assumption that the GPG keys are on this pool.

**Comment by joepvd** [comment]
https://staging.travis-ci.org/travis-ci/travis_staging_test/builds/699078
https://staging.travis-ci.org/travis-ci/travis_staging_test/jobs/699082 is a sudo precise job.
This has not been tested for the Enterprise image.

**Pull request merged**